### PR TITLE
Add eval-only global mean relaxation to SingleModuleStep

### PR DESCRIPTION
**Pull request by mcgibbon**
Adds an eval-only Newtonian relaxation of the global mean of selected output variables toward configured target values, applied between any global-mean-removal inverse transform and the corrector so the corrector operates on the relaxed field. Eval/train state on the step is made explicit via new `train()`/`eval()` methods on `StepABC`, matching the `torch.nn.Module` API.

Changes:
- `fme.core.step.global_mean_relaxation`: new `GlobalMeanRelaxationConfig`, `GlobalMeanRelaxationVariableConfig`, and `GlobalMeanRelaxation`; per variable, subtracts `(area_weighted_mean(x) - target) / timescale_steps`. `target` may be a float or `"mean"` (resolved to the network normalizer's mean at build time).
- `fme.core.step.single_module.SingleModuleStepConfig`: new optional `global_mean_relaxation` field, validated against `out_names`. Wired into `step_with_adjustments` via a new kwarg, gated on the step's training flag so it is inert during training.
- `fme.core.step.step.StepABC`: new `train(mode=True)`/`eval()` methods and a `_training` flag, mirroring `torch.nn.Module`. `fme.ace.stepper.single_module.SingleModuleStepper.set_eval`/`set_train` now delegate to the step's methods instead of iterating modules directly.

- [x] Tests added
- [ ] If dependencies changed, "deps only" image rebuilt and "latest_deps_only_image.txt" file updated

### CODE REVIEW

File: fme/core/step/global_mean_relaxation.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
+import dataclasses
+from typing import Literal
+
+import torch
+
+from fme.core.gridded_ops import GriddedOperations
+from fme.core.normalizer import StandardNormalizer
+from fme.core.typing_ import TensorDict, TensorMapping
+
+USE_NORMALIZATION_MEAN: Literal["mean"] = "mean"
```
Comment (yyexela) [inline fme/core/step/global_mean_relaxation.py:10]: Question: Is this the right name to use? The name `USE_NORMALIZATION_MEAN` seems like it'd be a bool more than a fixed literal string.

File: fme/core/step/step.py [hunk 2]
```diff
@@ -235,11 +235,33 @@ def from_state(cls, state: Mapping[str, Any]) -> Self:
 class StepABC(abc.ABC):
     SelfType = TypeVar("SelfType", bound="StepABC")
 
+    def __init__(self) -> None:
+        # Mirrors ``torch.nn.Module.training`` so that step-level eval/train
+        # state is observable without reaching into the underlying modules.
+        self._training: bool = True
```
Comment (yyexela) [inline fme/core/step/step.py:241]: Claude: This new base-class `__init__` sets `_training`, but `MockStep` in
  `test_step_registry.py` doesn't call `super().__init__()`, so it won't have
  that attribute. Calling `.train()` or `.eval()` on a `MockStep` will raise
  `AttributeError`. Fix: add `super().__init__()` as the first line of
  `MockStep.__init__`.

Comment (yyexela) [reply fme/core/step/step.py:241]: Alexey: If this is right, also add a test for this

File: fme/core/step/global_mean_relaxation.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+class GlobalMeanRelaxationConfig:
+    """
+    Newtonian relaxation of the global mean of selected output variables
+    toward configured target values.
+
+    The relaxation is applied during ``.step`` in eval mode only, after
+    the network call (and any global-mean-removal inverse transform) and
+    before the corrector. For each named variable, the step subtracts a
+    uniform offset of ``(area_weighted_mean(x) - target) / timescale_steps``
+    from the field.
```
Comment (yyexela) [inline fme/core/step/global_mean_relaxation.py:53]: Issue: Missing integration test for global_mean_removal + global_mean_relaxation together

Suggestion: Make sure the ordering is tested since that's a big part of how the PR says the data should flow.

File: fme/core/step/global_mean_relaxation.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        self,
+        targets: dict[str, float],
+        timescales: dict[str, float],
+        gridded_operations: GriddedOperations,
+    ):
+        if set(targets) != set(timescales):
+            raise ValueError(
+                "targets and timescales must have the same keys: "
+                f"targets={list(targets)}, timescales={list(timescales)}."
+            )
```
Comment (yyexela) [inline fme/core/step/global_mean_relaxation.py:130]: Issue: This guard is never tested.

Suggestion: Looks like `class GlobalMeanRelaxation` only ever gets made in `GlobalMeanRelaxationConfig.build()`, so this guard would never trigger since in that case `targets` == `timescales`. Either test this guard or make `class GlobalMeanRelaxation` private (ie. `class _GlobalMeanRelaxation`)

File: fme/core/step/single_module.py [hunk 5]
```diff
@@ -401,6 +421,16 @@ def network_call(input_norm: TensorDict) -> TensorDict:
...
 
+        # Newtonian relaxation toward configured target global means is an
+        # eval-time-only adjustment; gating on the step's training flag
+        # keeps it from contributing to the loss during training.
+        if self._global_mean_relaxation is not None and not self._training:
+            global_mean_relaxation: GlobalMeanRelaxation | None = (
+                self._global_mean_relaxation
+            )
+        else:
+            global_mean_relaxation = None
```
Comment (yyexela) [inline fme/core/step/single_module.py:432]: Nitpick: This can be simplified to:


```
global_mean_relaxation = self._global_mean_relaxation if not self._training else None
```

Comment (yyexela) [reply fme/core/step/single_module.py:432]: Pretty ugly otherwise

File: fme/core/step/global_mean_relaxation.py (hunk 1 above)
Comment (yyexela) [reply fme/core/step/global_mean_relaxation.py:10]: Suggestion: Use `NORMALIZATION_MEAN_TARGET` instead

**Review by yyexela — COMMENTED** [review]
Looks good overall, just some comments/questions/suggestions to clarify.
